src/compiler/program.ts

```diff
@@ -2426,6 +2433,7 @@ namespace ts {
         switch (extension) {
             case Extension.Ts:
             case Extension.Dts:
+            case Extension.Json:
                 // These are always allowed.

```

Comment:
Assuming we would only ever *get* a `.json` module if the option was set? Maybe could put that in the comment.